spirv_glsl.cpp

```diff
@@ -2796,7 +2797,13 @@ void CompilerGLSL::emit_texture_op(const Instruction &i)
 		break;
 	}
 
-	auto &imgtype = expression_type(img);
+	// Bypass pointers because we need the real image struct
+	auto &type = expression_type(img);
+	auto &imgtype = get<SPIRType>(type.self);
+
+	// Mark that this shader reads from this image
+	((SPIRType &)imgtype).image.is_read = true;
```

Comment:
Odd. I was getting a const cast error earlier...so that cast was to remove the const.

It is not happening now...so I've removed the messy cast.